tempesta_fw/classifier/frang.c

```diff
@@ -502,14 +537,38 @@ frang_http_host_check(const TfwHttpReq *req, FrangAcc *ra)
 	return ret;
 }
 
+static int
+frang_resp_time_in_frame(const int ta, const int tb)
+{
+	return tb - ta + (ta > tb ? FRANG_RESP_TIME_PERIOD : 0)
+		< frang_cfg.http_resp_code_block->tf;
+}
+
+static int
+frang_bad_resp_limit(FrangAcc *ra)
+{
+	FrangRespCodeStat *stat = ra->resp_code_stat;
+	long cnt = 0;
+	const int ts = (jiffies / HZ) % FRANG_RESP_TIME_PERIOD;
+	int i = 0;
+
+	for (; i < FRANG_FREQ; ++i) {
+		if (frang_resp_time_in_frame(stat[i].ts, ts))
+			cnt += stat[i].cnt;
+		if (cnt > frang_cfg.http_resp_code_block->limit)
```

Comment:
I see an optimisation effort here, but the condition is called for every `stat[X]`. Either move it to previous `if` statement body or after the cycle end. The array is small, the optimisation is not required here.